ArduCopter/avoidance_handler__avoid.cpp

```diff
@@ -0,0 +1,43 @@
+#include "avoidance_handler.h"
+
+#include <AP_Notify/AP_Notify.h>
+#include "Copter.h"
+
+bool AvoidanceHandler__AVOID::update()
+{
+    if (!AvoidanceHandler__ModeChange::update()) {
+        return false;
+    }
+
+    // only update our destination once per second
+    uint32_t now = AP_HAL::millis();
+    if (now - _last_wp_update > 1000) {
+        _last_wp_update = now;
+        if (!do_avoid_navigation()) {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+
+bool AvoidanceHandler__AVOID::do_avoid_navigation()
+{
+    Vector3f dest;
+    if (!new_destination(dest)) {
+        return false;
+    }
+    ::fprintf(stderr, "Set new destination (%f %f %f)\n",
```

Comment:
any reason you're not using a GCS sendtext for this? Debug only I presume but it looks like useful data